signature/policy_eval_signedby.go

```diff
@@ -47,20 +47,22 @@ func (pr *prSignedBy) isSignatureAuthorAccepted(image types.UnparsedImage, sig [
 		return sarRejected, nil, err
 	}
 	defer os.RemoveAll(dir)
-	mech, err := newGPGSigningMechanismInDirectory(dir)
-	if err != nil {
-		return sarRejected, nil, err
-	}
+	/*
+		mech, err := newGPGtypes.SigningMechanismInDirectory(dir)
+		if err != nil {
+			return sarRejected, nil, err
+		}
+	*/
```

Comment:
Moving the mechanism selection/creation out of `prSignedBy` really won’t work:

* The mechanism in principle depends on `pr.KeyType`; a single `PolicyContext.IsRunningImageAllowed` may involve policies with various `prSignedBy` instances with different `pr.KeyType` values. _Right now_ we only support `SBKeyTypeGPGKeys` but the code above already anticipates other values.

* With keys specified inside the policy as `pr.KeyPath`/`pr.KeyData`, we really want a temporary instance of the mechanism with its own state; the caller shouldn’t have to deal with creating individual temporary directories.

AFAICT the easiest way to do this is to have the two `SigningMechanism` implementations available in a single package, chosen by a build tag. Then nobody else needs to care.

Another possibility would be to use `types.SystemContext`, as we do for other environment overrides (and `signature.NewPolicy.Context(ctx *types.SystemContext, policy *Policy)` copying the `ctx` reference into the returned `PolicyContext`); but `types.SystemContext` is designed to always have a default, so we would need to build a reference to one of the implementations into `signature` anyway… and then we are more or less back in the build tag area.

@runcom is there another approach I am missing?

(I’d be happy to work on this, but I do need you to ACK that the chosen approach is usable in OpenShift.)